File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:

        
        def dfs(node,curr_max):


            if not node:

                return 0
            cnt = 0
            if node.val >= curr_max:

                cnt = 1

                curr_max = node.val

            cnt += dfs(node.left, curr_max)
            cnt += dfs(node.right , curr_max)


            return cnt

        return dfs(root,root.val)




        




        

File: Data Structures & Algorithms/palindromic-substrings/submission-1.go
func countSubstrings(s string) int {
	n := len(s) 

	cnt := 0

	var expand func(left, right int)  


	expand = func(left, right int)  {
		

		for left >= 0 && right < n && s[left] == s[right] {
			cnt++
			left--
			right++
		}


	}

	for i := 0; i < n; i++ {

		expand(i, i)
		expand(i, i+1)
		
	}
	return cnt    
}
